### PR TITLE
Updated Ambassador Ingress Controller to version 7.2.2

### PR DESCRIPTION
**Pull request by vladciobancai**
## BACKGROUND
* Upgraded Ambassador to version 7.2.2
* Fixing https://github.com/digitalocean/marketplace-kubernetes/issues/316
-----------------------------------------------------------------------

## Changes
* Updated deploy script to use the ambassador version 7.2.2
* Removed unused default values from the values.yml 
* Update README with the steps

-----------------------------------------------------------------------

## Checklist
- [x] review the [contributing doc](https://github.com/digitalocean/marketplace-kubernetes/blob/master/CONTRIBUTING.md) with steps for both adding or updating your application (if applicable)
------------------------------------------------------------------------

Reviewer: @marketplace-eng

### CODE REVIEW

File: stacks/ambassador/deploy.sh [hunk 1]
```diff
@@ -5,15 +5,15 @@ set -e
 ################################################################################
 # repo
 ################################################################################
-helm repo add datawire https://getambassador.io
-helm repo update > /dev/null
+helm repo add datawire https://app.getambassador.io
+helm repo update datawire > /dev/null
```
Comment (caquillo07) [inline stacks/ambassador/deploy.sh:9]: should we specific a specific minimum Helm 3 version, or encourage to use latest 3? I tried to run this on my system using helm 3.6.x and it failed to install saying `helm repo update does take any parameters`. I upgraded to 3.8.x and it worked then

**Review by caquillo07 — COMMENTED** [review]
This looks good, just a few findings when testing myself.

1. I noticed some of the scripts are missing executable permissions, not sure if this is expected, but wanted to bring it up as a just in case 

```shell
➜  marketplace-kubernetes-upgrade_ambassador_stack git:(master) ✗ ll ./stacks/ambassador/
drwxrwxr-x hector staff 256 B  Tue Feb 15 11:25:39 2022  .
drwxrwxr-x hector staff 1.5 KB Tue Feb 15 11:25:39 2022  ..
.rw-rw-r-- hector staff  48 KB Tue Feb 15 11:25:39 2022  Ambassador_Transparent.svg
.rwxr-xr-x hector staff 1.1 KB Tue Feb 15 11:25:39 2022  deploy.sh
.rw-rw-r-- hector staff 6.2 KB Tue Feb 15 11:25:39 2022  README.md
.rw-rw-r-- hector staff 286 B  Tue Feb 15 11:25:39 2022  uninstall.sh
.rw-rw-r-- hector staff 934 B  Tue Feb 15 11:25:39 2022  upgrade.sh
.rw-rw-r-- hector staff 168 B  Tue Feb 15 11:25:39 2022  values.yml
```

2. Seems like we may need to pin this to a specific minimum version of helm, or recommend the latest as this did not work fro me using version 3.6.x and that could be confusing to users.
3. I tried to access the installation using the url presented in the welcome message `https://SERVICE_IP/edge_stack_admin/#dashboard` and that just gave me connection errors, even after 15 mins of the service being up.  Hitting the IP using HTTP returned an empty response. This may not be an installation specific issue, but wanted to bring that up as well.

Everything else looks good, waiting for you input here before approving

File: stacks/ambassador/README.md [hunk 2]
```diff
@@ -1,152 +1,82 @@
...
 
-### Upgrading
+followed by:
 
-To upgrade Ambassador Edge Stack, use the following:
-```
-kubectl apply -f https://www.getambassador.io/yaml/aes-crds.yaml
-helm repo update && helm upgrade -n ambassador ambassador datawire/ambassador
+```bash
+kubectl delete ns ambassador
```
Comment (caquillo07) [inline stacks/ambassador/README.md:72]: I noticed that the namespace `emissary-system` is also created as part of this install, should we add this one to this as well?

File: stacks/ambassador/deploy.sh (hunk 1 above)
Comment (vladciobancai) [reply stacks/ambassador/deploy.sh:9]: I have removed `datawire` from the command `helm repo update > /dev/null` to be compatible with all helm 3 versions

File: stacks/ambassador/README.md (hunk 2 above)
Comment (vladciobancai) [reply stacks/ambassador/README.md:72]: I have updated the uninstall part to clean up the namespace `emissary-system` but also the crd's that were installed

**Comment by vladciobancai** [comment]
> This looks good, just a few findings when testing myself.
> 
> 1. I noticed some of the scripts are missing executable permissions, not sure if this is expected, but wanted to bring it up as a just in case
> 
> ```shell
> ➜  marketplace-kubernetes-upgrade_ambassador_stack git:(master) ✗ ll ./stacks/ambassador/
> drwxrwxr-x hector staff 256 B  Tue Feb 15 11:25:39 2022  .
> drwxrwxr-x hector staff 1.5 KB Tue Feb 15 11:25:39 2022  ..
> .rw-rw-r-- hector staff  48 KB Tue Feb 15 11:25:39 2022  Ambassador_Transparent.svg
> .rwxr-xr-x hector staff 1.1 KB Tue Feb 15 11:25:39 2022  deploy.sh
> .rw-rw-r-- hector staff 6.2 KB Tue Feb 15 11:25:39 2022  README.md
> .rw-rw-r-- hector staff 286 B  Tue Feb 15 11:25:39 2022  uninstall.sh
> .rw-rw-r-- hector staff 934 B  Tue Feb 15 11:25:39 2022  upgrade.sh
> .rw-rw-r-- hector staff 168 B  Tue Feb 15 11:25:39 2022  values.yml
> ```

Updated all the bash scripts to be executable now

> 2. Seems like we may need to pin this to a specific minimum version of helm, or recommend the latest as this did not work fro me using version 3.6.x and that could be confusing to users.

Updated the helm update command to be supported by all helm 3 versions 

> 3. I tried to access the installation using the url presented in the welcome message `https://SERVICE_IP/edge_stack_admin/#dashboard` and that just gave me connection errors, even after 15 mins of the service being up.  Hitting the IP using HTTP returned an empty response. This may not be an installation specific issue, but wanted to bring that up as well.
> 
> Everything else looks good, waiting for you input here before approving

The Ambassador 2.X doesn't comes with some default Services (eg Listener, Mapping and Host) already installed, they need to be created manually before the ingress controller can be used.  For more details https://www.getambassador.io/docs/edge-stack/latest/about/changes-2.x/#2-listeners-hosts-and-mappings

**Comment by caquillo07** [comment]
> > This looks good, just a few findings when testing myself.
> > 
> > 1. I noticed some of the scripts are missing executable permissions, not sure if this is expected, but wanted to bring it up as a just in case
> > 
> > ```shell
> > ➜  marketplace-kubernetes-upgrade_ambassador_stack git:(master) ✗ ll ./stacks/ambassador/
> > drwxrwxr-x hector staff 256 B  Tue Feb 15 11:25:39 2022  .
> > drwxrwxr-x hector staff 1.5 KB Tue Feb 15 11:25:39 2022  ..
> > .rw-rw-r-- hector staff  48 KB Tue Feb 15 11:25:39 2022  Ambassador_Transparent.svg
> > .rwxr-xr-x hector staff 1.1 KB Tue Feb 15 11:25:39 2022  deploy.sh
> > .rw-rw-r-- hector staff 6.2 KB Tue Feb 15 11:25:39 2022  README.md
> > .rw-rw-r-- hector staff 286 B  Tue Feb 15 11:25:39 2022  uninstall.sh
> > .rw-rw-r-- hector staff 934 B  Tue Feb 15 11:25:39 2022  upgrade.sh
> > .rw-rw-r-- hector staff 168 B  Tue Feb 15 11:25:39 2022  values.yml
> > ```
> 
> Updated all the bash scripts to be executable now
> 
> > 2. Seems like we may need to pin this to a specific minimum version of helm, or recommend the latest as this did not work fro me using version 3.6.x and that could be confusing to users.
> 
> Updated the helm update command to be supported by all helm 3 versions
> 
> > 3. I tried to access the installation using the url presented in the welcome message `https://SERVICE_IP/edge_stack_admin/#dashboard` and that just gave me connection errors, even after 15 mins of the service being up.  Hitting the IP using HTTP returned an empty response. This may not be an installation specific issue, but wanted to bring that up as well.
> > 
> > Everything else looks good, waiting for you input here before approving
> 
> The Ambassador 2.X doesn't comes with some default Services (eg Listener, Mapping and Host) already installed, they need to be created manually before the ingress controller can be used. For more details https://www.getambassador.io/docs/edge-stack/latest/about/changes-2.x/#2-listeners-hosts-and-mappings

Thanks for clarifying!

**Review by caquillo07 — COMMENTED** [review]
Looking great, thanks for doing this work!

**Review by caquillo07 — APPROVED** [review]

**Pull request merged**